services/core-web/common/components/tailings/TailingsSummaryPage.tsx

```diff
@@ -75,222 +77,246 @@ interface TailingsSummaryPageProps {
   clearTsf?: typeof clearTsf;
   isDirty?: (form: string) => boolean;
   initialValues?: Partial<ITailingsStorageFacility>;
+  userRoles?: string[];
+  userAction: string;
 }
 
 export const TailingsSummaryPage: FC<InjectedFormProps<ITailingsStorageFacility> &
   TailingsSummaryPageProps> = (props) => {
-    const { mines, history, formErrors, formValues, mineGuid, tsfGuid, tab } = props;
-    const [isLoaded, setIsLoaded] = useState(false);
-    const [isReloading, setIsReloading] = useState(false);
-    const [uploadedFiles, setUploadedFiles] = useState([]);
+  const { mines, history, formErrors, formValues, mineGuid, tsfGuid, tab, userAction } = props;
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [isReloading, setIsReloading] = useState(false);
+  const [uploadedFiles, setUploadedFiles] = useState([]);
+  const [canEditTSF, setCanEditTSF] = useState(false);
+  const { renderConfig, components, routes, isCore } = useContext(TailingsContext);
+  const { Loading } = components;
+  const action = userAction ? userAction : "edit";
```

Comment:
set defaults when destructuring props like userAction="edit" on line 86.

Reply:
This has been adjusted now.